src/Diagram.hs

```diff
@@ -6,40 +6,43 @@ module Diagram where
 import Algebra.Graph.Labelled as LG
 import Control.Lens.TH (makeLenses)
 import FlowChart
+import GanttChart
 import Parser
-import Text.Megaparsec hiding (State)
 import Text.Megaparsec.Char (string)
 import qualified Text.Megaparsec.Char.Lexer as L
 
 data Diagram
   = FlowChart
       { _orientation :: Orientation,
-        _graph :: FlowChartGraph
+        _flowGraph :: FlowChartGraph
       }
-  | Others
+  | GanttChart {_ganttGraph :: GanttChartGraph}
   deriving (Eq, Show, Generic)
 
 makeLenses ''Diagram
 
-data DiagramType = FlowChartType | OtherType deriving (Eq, Show, Generic)
+pDiagram :: Parser [Diagram]
+pDiagram = many $ pGanttChartDiagram <|> pFlowChartDiagram
 
-makeLenses ''DiagramType
-
-pDiagram :: Parser Diagram
-pDiagram = L.nonIndented sc (L.indentBlock sc p)
+pGanttChartDiagram :: Parser Diagram
```

Comment:
This should be `pGanttChartDiagram :: Parser GanttChartGraph`, and `pDiagram` should just `fmap D_GanttChart` over the result. 

In general, `pFoo :: Parser Foo`.

Reply:
I made some changes and am still thinking how to do it better here. May I have your advice?

Reply:
pDiagram looks good now.